src/modules/services/keycloak.nix

```
@@ -0,0 +1,498 @@
+{ config
+, lib
+, pkgs
+, ...
+}:
+
+let
+  cfg = config.services.keycloak;
+
+  inherit (lib)
+    mkIf
+    mkMerge
+    mkOption
+    mkRenamedOptionModule
+    mkPackageOption
+    types
+    ;
+
+  inherit (types)
+    nullOr
+    oneOf
+    listOf
+    attrsOf
+    ;
+
+  assertStringPath =
+    optionName: value:
+    if builtins.isPath value then
+      throw ''
+        services.keycloak.${optionName}:
+          ${builtins.toString value}
+          is a Nix path, but should be a string, since Nix
+          paths are copied into the world-readable Nix store.
+      ''
+    else
+      value;
+in
+{
+  imports = [
+    (mkRenamedOptionModule [ "keycloak" "enable" ] [ "services" "keycloak" "enable" ])
+  ];
+
+  options.services.keycloak = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = ''
+        Whether to enable the Keycloak identity and access management
+        server.
+      '';
+    };
+
+    sslCertificate = mkOption {
+      type = nullOr types.path;
```

```suggestion
      type = nullOr (types.pathWith { inStore = false; absolute = true; });
```

This gets rid of `assertStringPath` helper